extensions/map-layers/src/test/UserPreferencesMock.ts

```diff
@@ -39,27 +41,42 @@ export function setup() {
       returnVal = iTwinPrefs.get(arg.key);
     }
 
+    if (undefined !== returnVal)
+      return returnVal;
+
+    if (!arg.key)
+      return Array.from(blankPrefs.values());
+    returnVal = blankPrefs.get(arg.key);
+
     return returnVal;
   };
 
   const deleteStub = async (arg: PreferenceKeyArg & ITwinIdArg & TokenArg) => {
-    if (undefined === iModelPrefs || undefined === iTwinPrefs)
+    if (undefined === iModelPrefs || undefined === iTwinPrefs || undefined === blankPrefs)
```

Comment:
Will a blank connection not have an iTwin Id though to associate it back to a project or other context?

Reply:
@calebmshafer 
yes I think you are right, there will always  be itwin ID with a blank connection, but I thought we could allow not passing it anyway; if the hosting application doesn't need it (i.e. saving the settings in the local storage).  If you prefer, I can keep the iTwin id mandatory.

Reply:
ok made iTwinId mandatory.

Reply:
@calebmshafer Please review one more time.

Reply:
@mdastous-bentley looks like the `replaceSource` method still has it as optional. See comment https://github.com/iTwin/itwinjs-core/pull/3256#discussion_r817113559

Reply:
@calebmshafer should be good now

Reply:
cool, looks good.